lib/smb2/client.rb

```diff
@@ -193,17 +190,19 @@ def send_recv(request)
     dispatcher.recv_packet
   end
 
-  # @param tree [String]
+  # Connect to a share
+  #
+  # @param tree [String] Something like "\\\\hostname\\tree"
   # @return [Smb2::Tree]
   def tree_connect(tree)
-    packet = Smb2::Packet::TreeConnectRequest.new do |request|
-      request.tree = tree.encode("utf-16le")
-    end
+    packet = Smb2::Packet::TreeConnectRequest.new(
+      tree: tree.encode("utf-16le")
```

Comment:
I like passing the tree in as a constructor argument over having to use a block.  Good change.